tests/structured_conf/test_structured_config.py

```diff
@@ -904,6 +904,12 @@ def test_str2str(self, module: Any) -> None:
         with raises(KeyValidationError):
             cfg[Color.RED]
 
+    def test_dict_subclass_data_preserved_upon_node_creation(self, module: Any) -> None:
+        src = module.DictSubclass.Str2Str()
+        src["baz"] = "qux"
+        cfg = OmegaConf.structured(src)
+        assert cfg.baz == "qux"
```

Comment:
If run against the master branch, the above test raises `ConfigAttributeError: Missing key baz`.
This PR addresses the error by inspecting for dict subclass data within the `get_attr_data` and `get_dataclass_data` functions.